src/services/TechRecordsService.ts

```diff
@@ -232,27 +232,72 @@ class TechRecordsService {
           throw new HTTPError(500, ERRORS.NO_UNIQUE_RECORD);
         }
-        const uniqueRecord = data[0];
+        const uniqueRecord = data[0];// call payload
         const oldTechRec = this.getTechRecordToArchive(uniqueRecord, statusCode);
-        const newRecord: any = _.cloneDeep(oldTechRec);
-        oldTechRec.statusCode = STATUS.ARCHIVED;
-        _.mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
-        this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
-        populateFields(newRecord);
-        uniqueRecord.techRecord.push(newRecord);
-        return uniqueRecord;
+        if(!newStatusCode) {
+          const newRecord: ITechRecord = cloneDeep(oldTechRec);
+          mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
+          this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
+          populateFields(newRecord);
+          uniqueRecord.techRecord.push(newRecord);
+          return uniqueRecord;
+        } else {
+
+          if(newStatusCode === STATUS.ARCHIVED) {
+            console.log("Should get here");
+            this.updateAtributesWhenNewStatusIsArchived(oldTechRec,msUserDetails);
+            console.log("New Status code: ",uniqueRecord.techRecord[0].statusCode);
+            return uniqueRecord;
+
+          } else if(newStatusCode === STATUS.CURRENT && oldTechRec.statusCode === STATUS.PROVISIONAL) {
+            if(uniqueRecord.techRecord.filter((techRecords: any) => {
+              return techRecords.statusCode === STATUS.CURRENT;}).length!==1) {
+              const newRecord: ITechRecord = cloneDeep(oldTechRec);
+              console.log("Makes the AC3 Val");
+              console.log("Old status code: ",oldTechRec.statusCode);
+              newRecord.statusCode = newStatusCode;
+              oldTechRec.statusCode = STATUS.ARCHIVED;
+              mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
+              this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
+              populateFields(newRecord);
+              uniqueRecord.techRecord.push(newRecord);
+              return uniqueRecord;
+            } else {
+              const currentrecord = this.getTechRecordToArchive(uniqueRecord, STATUS.CURRENT);
+              this.updateAtributesWhenNewStatusIsArchived(currentrecord,msUserDetails);
+              const newRecord: ITechRecord = cloneDeep(oldTechRec);
+              newRecord.statusCode = newStatusCode;
+              mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
+              this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
+              oldTechRec.statusCode = STATUS.ARCHIVED;
+              populateFields(newRecord);
+              uniqueRecord.techRecord.push(newRecord);
+              return uniqueRecord;
+            }
+          } else if(newStatusCode === STATUS.PROVISIONAL && oldTechRec.statusCode === STATUS.CURRENT) {
+            throw Error;
```

Comment:
This error needs to have a statusCode and a body in which you specify the error message